utils/diff/diff.go

```diff
@@ -16,6 +17,7 @@ import (
 // string into the dst string.
 func Do(src, dst string) (diffs []diffmatchpatch.Diff) {
 	dmp := diffmatchpatch.New()
+	dmp.DiffTimeout = time.Hour  // the default is time.Second which may be too little under heavy load
```

Comment:
Well, maybe we can do this an argument?

Maybe `DoWithTimeout`?

Reply:
Then we should remove the current `Do` function, because the root of the problem is a too small default timeout. wdyt?

Reply:
Nope, first we need to preserve the API, and second, I am not saying that we are going to keep the second, just that should be configurable.

Reply:
So I should add `DoWithTimeout` and also preserve the changed default timeout, correct?

Reply:
Yup